src/implementations/astroport/staking.rs

```diff
@@ -51,7 +51,17 @@ impl Stake for AstroportStaking {
 }
 
 impl Rewards for AstroportStaking {
-    fn claim_rewards(&self, _deps: Deps, _env: &Env) -> Result<Response, CwDexError> {
+    fn claim_rewards(&self, deps: Deps, env: &Env) -> Result<Response, CwDexError> {
+        let claimable_rewards: AssetList =
+            self.query_pending_rewards(&deps.querier, &env.contract.address)?;
+
+        let event =
+            Event::new("apollo/cw-dex/claim_rewards").add_attribute("type", "astroport_staking");
+
+        if claimable_rewards.len() == 0 {
```

Comment:
This currently won't work since the AssetList may contain assets with 0 amount. Lets keep this logic here and change `query_pending_rewards` to do a purge or similar.